openedx/core/djangoapps/content_libraries/docs/decisions/0004-upstream-block.rst

```diff
@@ -0,0 +1,89 @@
+4. Persist the relationship between blocks and their upstream source
+####################################################################
```

Comment:
@ormsbee @bradenmacdonald Generally, would you guys like to see this system kept within edx-platform, or built into the XBlock framework?

My initial instinct was to add the `upstream_block[_version]` fields to AuthoringMixin since only CMS needs to worry about synchronizing updates. But, both CMS and LMS will need to understand that `upstream_block.{}` values are to be interpreted as defaults, so the concept of "upstream" will bleed out of CMS in at least that sense.

Still, just like we do with Inheritance, we could keep this within edx-platform by making a special `UpstreamDefaultsFieldData` and `UpstreamDefaultsKeyValueStore` classes. Or, we could bake upstream defaults into XBlock. I'm leaning the latter, personally (although I would still like to keep the upstream-sync logic isolated into a corner of CMS).